### DISCUSSION TITLE
Improve the inference of the `primary_key`

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#165] [Feedback & Feature Proposal]
Currently, to infer the primary_key, the core engine chooses the first attribute found in the first document that contains `id` as the primary key.

## Problem

A document with `region_id` first and then `id` in the same document could not be inferred correctly without specifying the primary key. Since `id` is the unique identifier of a DB in most of the datasets, we might want to find a solution for this kind of dataset.

## Solution

An alternative could be to
- first, check if an attribute named exactly `id` exists and take it as primary_key
- if not, do the inference as we do currently

**Comment by gmourier** [comment]
Hello @curqui!

Indeed I think it's a good idea to have a dedicated field for id inference without taking into account the position of the field in the sent payloads. 

Typically `id` or `_id`.

We could also generate this unique `id` when it is not filled by the user so that he can still use the search engine in optimal conditions.

Selecting the first field containing `id` within the first document to make it the primary key is not the best solution and may prevent the indexing of all documents. (I think of the possible collision on `region_id` which is not unique among all documents.  If this one is chosen by the engine to be the primary key by the inference system some documents may not be indexed).

**Comment by gmourier** [comment]
Locked in favor of https://github.com/meilisearch/product/discussions/206
